crates/pathfinder/src/cairo/ext_py/service.rs

```diff
@@ -37,6 +41,31 @@ pub async fn start(
     let (child_shutdown_tx, _) = broadcast::channel(1);
     let command_rx: SharedReceiver<(Command, tracing::Span)> = Arc::new(Mutex::new(command_rx));
 
+    let processes_launched = metrics::register_counter!(METRIC_LAUNCHED_PROCESSES);
+    metrics::describe_counter!(
+        METRIC_LAUNCHED_PROCESSES,
+        metrics::Unit::Count,
+        "number of launched python subprocesses; equals sum of exited and failed subprocesses."
+    );
+
+    // exposing the Option<ExitStatus>: opaque, exit reason counts why our code ended up reacting
+    // like this. failed variant catches all errors.
+    for reason in super::SubprocessExitReason::all_labels() {
+        metrics::register_counter!(METRIC_EXITED_PROCESSES, "reason" => reason);
+    }
+    metrics::describe_counter!(
+        METRIC_EXITED_PROCESSES,
+        metrics::Unit::Count,
+        "number of normally exited subprocesses."
+    );
+
+    let processes_failed = metrics::register_counter!(METRIC_FAILED_PROCESSES);
+    metrics::describe_counter!(
+        METRIC_FAILED_PROCESSES,
+        metrics::Unit::Count,
+        "number of abnormally, due to bug, exited subprocesses."
+    );
+
```

Comment:
Maybe it's worth wrapping any counter registering into a separate `fn`.

Reply:
Why?

Reply:
To shorten the function body and keep its readability at the current level.

Reply:
Do you like it now?